vignettes/reactlog.Rmd

```diff
@@ -0,0 +1,346 @@
+
+```{r}
+knitr::include_graphics("images/step_information.png")
+```
+
+The current step information is displayed above the progress bar just after the navigation buttons. For each step within the reactlog, corresponding information is displayed.
+
+To the left of the search bar is the Time and Session information.  The Time will display the relative time difference from the first recorded step.  The Session will display the Shiny session that the current step is being executed in.
+
+##### Search Bar
+
+The search bar is a live search bar that requires a minimum of three characters to begin searching.  All matches are found against the name or ID of each reactive graph element.  If no matches are found, the full graph is displayed.  All matching elements are increased in size to signify a match.  When searching for an element, all matching elements' family trees will be displayed.
+
+Searching can be very useful to filter a large graph down to only the reactive elements involved in a calculation.
+
+```{r, out.width="40%"}
+knitr::include_graphics("images/search_bar.png")
+```
+
+> Pro tip: Use partial matching to filter the graph down to a particular module name.  The search above would look at the `input` values in the _details_ module.
+
+
+
+### Dependency Graph
+
+Once a reactlog has been recorded, all possible dependency connections are calculated to visually display which reactive elements are connected other reactive elements.  These possible connections will be displayed at all times to maintain graph structure and identify which elements will depend on other elements.  The directed, acyclic (typically) graph lays out the logical flow from left to right how reactive dependencies are invalidated and calculated.
+
+
+#### Reactive Graph Nodes
+
+```{r}
+knitr::include_graphics("images/elements.png")
+```
+
+Reactive inputs are values that are the beginning of a reactive chain.  They are the leftmost starting points of the reactive graph.  Reactive sources are typically values created in `input`, `clientData`, `shiny::reactiveVal`, or `shiny::reactiveValues`.
+
+Reactive expressions are conductors between sources and enpoints.  Their values depend on other reactive elements and are depended on by other reactive elements.  These expressions help with intermediate calculations, preventing duplicate calculations, and aid in logical flow.
+
+Reactive endpoints are values that are being stored to the `output` object or are `observe`ing other reactive elements to produce side effects.
+
+> Pro tip: Use the `label` parameter when defining reactive elements.  This will help your debug process and avoid long, code-exposing reactive element labels.
+
```

Comment:
Are you referring to the `label` argument of `observe()`? Is there anyway to consistently label outputs?

Reply:
there is a label arg for reactive and observe.  

> anyway to consistently label outputs?
Without being manual? no.  Shiny takes the first part of the body of the expression and makes that the label